app/models/host.rb

```diff
@@ -1939,17 +1939,15 @@ def self.node_types
   end
 
   def self.openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    ems.empty? ? false : Host.where(:ems_id => ems).exists?
+    ManageIQ::Providers::Openstack::InfraManager::Host.count > 0
   end
 
   def self.non_openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    Host.where.not(:ems_id => ems).exists?
+    (Host.count - ManageIQ::Providers::Openstack::InfraManager::Host.count) > 0
```

Comment:
@kbrock I looked badly, your code is actually equivalent to what @bdune has there, just doing it in one query

btw. should be ems = ManageIQ::Providers::Openstack::InfraManager.all